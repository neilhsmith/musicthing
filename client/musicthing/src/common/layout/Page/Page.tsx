import { ReactNode } from "react";
import styled from "styled-components";

/**
 * A basic component used to wrap individual pages. It simply provides a colors,
 * sets its height to be equal to the viewport and renders children.
 */

type PageProps = {
  /** The children to render within the page. */
  children: ReactNode;
};

const StyledPage = styled.div`
  height: 100vh;
  background-color: ${(props) => props.theme.colors.background1};
  color: ${(props) => props.theme.colors.text};
`;

const Page = ({ children }: PageProps) => <StyledPage>{children}</StyledPage>;

export default Page;
