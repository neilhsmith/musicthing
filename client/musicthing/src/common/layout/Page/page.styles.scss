.page {
  height: 100%;
  background-color: var(--background-color-1);
  color: var(--text-color);
}
