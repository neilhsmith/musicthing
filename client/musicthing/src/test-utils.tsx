import React, { ReactElement } from "react";
import { render, screen } from "@testing-library/react";
import { ThemeProvider } from "styled-components";

import { dark } from "styles/themes";

const AllProviders = ({ children }: any) => {
  return <ThemeProvider theme={dark}>{children}</ThemeProvider>;
};

const customRender = (ui: ReactElement, options?: any) =>
  render(ui, { wrapper: AllProviders, ...options });

export * from "@testing-library/react";

export { customRender as render, screen, dark as defaultTheme };
