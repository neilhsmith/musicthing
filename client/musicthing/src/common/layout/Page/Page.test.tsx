import { render, screen, defaultTheme } from "test-utils";

import Page from "./Page";

describe("Component: Page", () => {
  it("matches its snapshot", () => {
    const { container } = render(
      <Page>
        <span data-testid="42">a child</span>
      </Page>
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it("fills the viewport", () => {
    const { container } = render(
      <Page>
        <span data-testid="42">a child</span>
      </Page>
    );

    expect(container.firstChild).toHaveStyleRule("height", "100vh");
  });

  it("sets theme's background1 as the background-color", () => {
    const { container } = render(
      <Page>
        <span data-testid="42">a child</span>
      </Page>
    );

    expect(container.firstChild).toHaveStyleRule(
      "background-color",
      defaultTheme.colors.background1
    );
  });

  it("sets theme's text as the color", () => {
    const { container } = render(
      <Page>
        <span data-testid="42">a child</span>
      </Page>
    );

    expect(container.firstChild).toHaveStyleRule(
      "color",
      defaultTheme.colors.text
    );
  });
});
