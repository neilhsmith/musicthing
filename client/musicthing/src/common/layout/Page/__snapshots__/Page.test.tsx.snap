// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Page matches its snapshot 1`] = `
.c0 {
  height: 100vh;
  background-color: #12181b;
  color: #c4c4c4;
}

<div
  class="c0"
>
  <span
    data-testid="42"
  >
    a child
  </span>
</div>
`;
